provisioning/miniprov/hare_mp/store.py

```diff
@@ -77,6 +83,23 @@ def __init__(self, url: str, index='hare'):
     def _raw_get(self, key: str) -> str:
         return self.conf.get(self.index, key)
 
+    def get_machine_ids_for_service(self, service_type: str) -> List[str]:
+        nodes: List[str] = []
+        machines: Dict[str, Any] = self.get('node')
+        services = self.search_val('node', 'services', service_type)
+        for machine_id in machines.keys():
+            result = [srv for srv in services if machine_id in srv]
+            if result:
+                nodes.append(machine_id)
+        return nodes
+
+    def get_hostnames_for_service(self, service_type: str) -> List[str]:
+        nodes: List[str] = []
```

Comment:
I think "nodes" need to be replaced here "hostnames" for better understanding

Reply:
Function name defined itself about the type of list. No need to modify variable name as such.

Reply:
Right. Actually from function name itself I added this comment to have variable name accordingly.

Reply:
Same as previous reply here.

Reply:
Skipping this for now as all testing is done.
This can be taken later.